vm/devices/get/guest_emulation_transport/src/client.rs

```diff
@@ -186,7 +154,9 @@ impl GuestEmulationTransportClient {
             capacity: response.capacity,
             bytes_per_logical_sector: response.bytes_per_logical_sector,
             bytes_per_physical_sector: response.bytes_per_physical_sector,
-            maximum_transfer_size_bytes: response.maximum_transfer_size_bytes,
+            maximum_transfer_size_bytes: response
+                .maximum_transfer_size_bytes
+                .min(get_protocol::MAX_PAYLOAD_SIZE as u32),
```

Comment:
is this the right abstraction level to put this clamp at?

Reply:
Hmm, fair point.

Maybe we should push this clamp into the process_loop itself? and probably warn if the host is misbehaving, and reporting an unexpected max transfer size?

Reply:
We could. The process loop's handling for this today is straight passthrough, though. I don't know the philosophy of when we do stuff in the process loop and when in the client.

Reply:
(A lot of validation/parsing code is in the client today.)

Reply:
Like many things in the GET, the design is a bit... iffy.

In the past, it made more sense to do a lot of validation/parsing in the client as we were handing out raw references to the GET client all over the place. It seems that nowadays, we do a better job "API slicing" the GET client into smaller subsets via wrapper objects, and its those objects that typically implement additional validation semantics.

But if you look at the code, you'll find plenty of exceptions.

I'll let you decide where you'd like to draw the line here - I don't think I have hard and fast guidance here at the moment.

Reply:
I'll keep it in the client for now, but add the warn.